plynx/cdp/handlers/webhook_handler.py

```diff
@@ -0,0 +1,130 @@
+            self.message.data[StreamDataFields.STREAM_QUEUE_ID],
+            self.message.data[StreamDataFields.INSTANCE_NAME],
+        )
+
+    def _retrieve_workflows(self):
+        """Query target workflows by given criterias."""
+
+        workflows = get_db_connector()[Collections.TEMPLATES].find(
+            {
+                "$and": [
+                    {"kind": NodeTypes.PYTHON_WORKFLOW},
+                    {"node_status": NodeStatus.CREATED},
+                    {
+                        "$and": [
+                            {
+                                "parameters.value.value": {
+                                    "$elemMatch": {
+                                        "parameters.name": StreamDataFields.DATA_SOURCE,
+                                        "parameters.value": DataSource.STREAM,
+                                    }
+                                }
+                            },
+                            {
+                                "parameters.value.value": {
+                                    "$elemMatch": {
+                                        "parameters.name": StreamDataFields.TOPIC,
+                                        "parameters.value": self.stream_data.topic,
+                                    }
+                                }
+                            },
+                            {
+                                "parameters.value.value": {
+                                    "$elemMatch": {
+                                        "parameters.name": StreamDataFields.INSTANCE_NAME,
+                                        "parameters.value": self.stream_data.instance_name,
+                                    }
+                                }
+                            },
+                        ]
+                    },
+                ],
+            }
+        )
+
+        return workflows
+
+    def process(self) -> None:
+        """Process the stream data event."""
+
+        workflows = self._retrieve_workflows()
+        for workflow in workflows:
+
+            workflow: WorkflowNode = Node.from_dict(workflow)
+            workflow_to_run = workflow.clone(NodeClonePolicy.NODE_TO_RUN)
+
+            ops_node: Parameter = workflow_to_run.parameters[0]
+            assert (
+                ops_node.parameter_type == "list_node"
+            ), "Parameter type `list_node` is missing in the workflow."
+
+            operations: List[OperationNode] = ops_node.value.value
+            for op in operations:
+
+                payload_updated = False
+                op_params: List[Parameter] = op.parameters
+
+                for p in op_params:
+                    if p.name == StreamDataFields.PAYLOAD:
+                        self.logger.info("Updating payload data with stream message.")
+
+                        p.value = self.stream_data.payload
+                        payload_updated = True
```

Comment:
```suggestion
                        break
```

No need to continue execution?

Reply:
good point, fixed